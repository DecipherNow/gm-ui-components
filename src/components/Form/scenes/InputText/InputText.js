import React from "react";
import PropTypes from "prop-types";

import InputLabel from "../../components/InputLabel";
import InputWrap from "../../components/InputWrap";
import InputHint from "../../components/InputHint";
import InputField from "../../components/InputField";

InputText.propTypes = {
  autoFocus: PropTypes.bool,
  defaultValue: PropTypes.string,
  disabled: PropTypes.bool,
  hint: PropTypes.string,
  label: PropTypes.string,
  maxLength: PropTypes.number,
  placeholder: PropTypes.string,
  readonly: PropTypes.bool
};

export default function InputText({
  autoFocus,
  defaultValue,
  readonly,
  disabled,
  hint,
  label,
  maxLength,
  placeholder,
  ...props
}) {
  return (
    <InputWrap>
      {label && <InputLabel>{label}</InputLabel>}
      <InputField
        type="text"
        autoFocus={autoFocus}
        defaultValue={defaultValue}
        disabled={disabled}
        readonly={readonly}
        maxLength={maxLength}
        placeholder={placeholder}
        {...props}
      />
      {hint && <InputHint>{hint}</InputHint>}
    </InputWrap>
  );
}
