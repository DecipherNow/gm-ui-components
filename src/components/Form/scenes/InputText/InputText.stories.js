import React from "react";
import { storiesOf } from "@storybook/react";
import { text, number, boolean } from "@storybook/addon-knobs/react";
import { withInfo } from "@storybook/addon-info";

import InputText from "./InputText";

const stories = storiesOf("Components|InputText", module);

stories.add(
  "default",
  withInfo("A text input component with label and optional hint.")(() => {
    return (
      <InputText
        autoFocus={boolean("autoFocus", false)}
        hint={text("hint", "hint")}
        label={text("label", "label")}
        defaultValue={text("defaultValue", "defaultValue")}
        placeholder={text("placeholder", "placeholder")}
        maxLength={number("maxLength", 25)}
      />
    );
  })
);
