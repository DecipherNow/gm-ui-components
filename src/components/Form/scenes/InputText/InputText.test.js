import React from "react";
import { shallow } from "enzyme";

import InputText from "./InputText";

describe("InputText", () => {
  it("matches snapshot", () => {
    const aInputText = shallow(
      <InputText label="Test" clickAction={() => {}} />
    );
    expect(aInputText).toMatchSnapshot();
  });
});
