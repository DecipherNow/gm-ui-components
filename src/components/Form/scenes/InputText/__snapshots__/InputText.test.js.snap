// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputText matches snapshot 1`] = `
<InputWrap>
  <InputLabelText>
    Test
  </InputLabelText>
  <InputField
    clickAction={[Function]}
    type="text"
  />
</InputWrap>
`;
